src/prefixmaps/datamodel/context.py

```diff
@@ -22,35 +24,76 @@
 
 class StatusType(Enum):
     """
-    Classification of prefix expansions
+    Classification of prefix expansions.
+
+    Note that only canonical mappings are exposed to the users of the library. However,
+    it can be useful for prefixmap ETL pipelines to include non-canonical mappings for
+    purposes of debugging.
     """
 
     canonical = "canonical"
+    """The canonical prefix expansion for a prefix. The set of all canonical mappings must be bijective."""
+
     prefix_alias = "prefix_alias"
+    """The prefix is an alias for an existing canonical prefix."""
+
     namespace_alias = "namespace_alias"
+    """The prefix is an alias for an existing canonical namespace."""
+
     multi_alias = "multi_alias"
+    """Both the prefix and the namespace are aliases for existing canonical namespaces."""
 
 
 @dataclass
 class PrefixExpansion:
     """
-    An individual mapping between a prefix and a namespace
+    An individual mapping between a prefix and a namespace.
+
+    A PrefixExpansion corresponds to a SHACL PrefixDeclaration (http://www.w3.org/ns/shacl#PrefixDeclaration)
```

Comment:
can you elaborate on what this is, since the link goes to a page with no information?

Reply:
From https://www.w3.org/TR/shacl/#dfn-prefix-declarations:

> The [values](https://www.w3.org/TR/shacl/#dfn-value) of the property sh:declare are [IRIs](https://www.w3.org/TR/shacl/#dfn-iri) or [blank nodes](https://www.w3.org/TR/shacl/#dfn-blank-node), and these values are called prefix declarations. The SHACL vocabulary includes the class sh:PrefixDeclaration as type for such [prefix declarations](https://www.w3.org/TR/shacl/#dfn-prefix-declarations) although no rdf:type triple is required for them. [Prefix declarations](https://www.w3.org/TR/shacl/#dfn-prefix-declarations) have exactly one value for the property sh:prefix. The values of sh:prefix are [literals](https://www.w3.org/TR/shacl/#dfn-literal) of datatype xsd:string. [Prefix declarations](https://www.w3.org/TR/shacl/#dfn-prefix-declarations) have exactly one value for the property sh:namespace. The values of sh:namespace are [literals](https://www.w3.org/TR/shacl/#dfn-literal) of datatype xsd:anyURI. Such a pair of values specifies a single mapping of a prefix to a namespace.

Reply:
this is a semantic URL like the URLs in RDF, RDFS, OWL, etc

You can fetch via curl

- `curl --header "Accept: text/turtle"  -L -s http://www.w3.org/ns/shacl\#PrefixDeclaration`

And the URL resolves to a page that contains a link for humans... not ideal. But anyway I think intended users of this library should be familiar with this